src/translation.cpp

```diff
@@ -0,0 +1,51 @@
+#include "translation.h"
+#include <QFile>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QDebug>
+
+Translation::Translation()
+{
+}
+
+void Translation::setTranslation(QLocale locale)
+{
+    auto defaultText = JsonFileToQMap(":/i18n/en.json");
+    if (defaultText.isEmpty()) {
+        throw std::runtime_error("Invalid translation file");
+    }
+    if (locale.bcp47Name() == "en") {
+        m_translations = defaultText;
+        return;
+    }
+    m_translations = JsonFileToQMap(":/i18n/" + locale.bcp47Name() + ".json");
+    for (auto &key : m_translations.keys()) {
+        if (m_translations.value(key).isEmpty()) {
+            m_translations.insert(key, defaultText.value(key));
+        }
+    }
```

Comment:
THere is very few chance the there is a key in the translation with an empty value. (but indeed we need to test that).
What we really want to check is if a key is missing. We should loop over the key of `defaultText` (and test the value in `m_translations`.